src/kakarot/library.cairo

```diff
@@ -355,54 +257,33 @@ namespace Kakarot {
         value: felt,
         data_len: felt,
         data: felt*,
-    ) -> (return_data_len: felt, return_data: felt*, success: felt) {
+    ) -> EVM.Summary* {
         alloc_locals;
-
-        if (to == 0) {
-            // TODO: read the nonce from the provided origin address, otherwise in view mode this will
-            // TODO: always use a 0 nonce
-            let (tx_info) = get_tx_info();
-            let (evm_contract_address) = CreateHelper.get_create_address(origin, tx_info.nonce);
-
-            let (success) = transfer(origin, evm_contract_address, value);
-            with_attr error_message(
-                    "Kakarot: eth_call: failed to transfer {value} tokens to {evm_contract_address}") {
-                assert success = TRUE;
-            }
-
-            let (starknet_contract_address, reverted) = deploy_contract_account(
-                origin=origin,
-                evm_contract_address=evm_contract_address,
-                value=value,
-                bytecode_len=data_len,
-                bytecode=data,
-            );
-            let (return_data) = alloc();
-            assert [return_data] = evm_contract_address;
-            assert [return_data + 1] = starknet_contract_address;
-            return (2, return_data, 1 - reverted);
+        let evm_contract_address = resolve_to(to, origin);
+        let (starknet_contract_address) = Accounts.compute_starknet_address(evm_contract_address);
+        tempvar address = new model.Address(starknet_contract_address, evm_contract_address);
+
+        let is_regular_tx = is_not_zero(to);
+        let is_deploy_tx = 1 - is_regular_tx;
+        let (bytecode_len, bytecode) = Accounts.get_bytecode(address.evm);
```

Comment:
won't this fail as the account isn't created yet?

Reply:
`get_bytecode` handles this and returns `(0, [])` if the account doesn't exist.